### DISCUSSION TITLE
A way to extend from a present .wsb

### DISCUSSION THREAD

**Discussion by freshgum-bubbles** [damienvanrobaeys/Run-in-Sandbox#35] [Ideas]
It would be nice if there was a way to also use a pre-existing `.wsb` configuration file for this app.

## Use-case

I have a .wsb that sets up a share between my Sandbox and host, along with a few other changes.
Currently, this app doesn't let me use that (not that I'm aware of anyway) :-(

**Comment by Rigavinil201** [comment]
None

**Reply by Rigavinil201** [reply]
Add

**Reply by freshgum-bubbles** [reply]
Not sure what you mean

**Reply by Joly0** [reply]
I mean, you can use existing .wsb files, but you would have to add the correct mounts for run-in-sandbox and the commands to do whatever you are trying to achieve.

But could you give us some more explanation of what your .wsb file is doing that you want to do with run-in-sandbox?

**Comment by freshgum-bubbles** [comment]
> But could you give us some more explanation of what your .wsb file is doing that you want to do with run-in-sandbox?

The main thing it does is opens up access to my Share folder, which is readable from the Sandbox.
It's really handy for somewhat maintaining persistence across Sandbox runs, as I can store portable 
programs in there to be executed on each run :-)

**Reply by Joly0** [reply]
I mean, you could use the "multiple apps" feature of run-in-sandbox for this. Though, i cant see alot of options on how we could acomplish this idea.
We could add another config for you too specify a path to add as a shared folder. Or maybe have a separate file with various shared folder mappings as a config file to read by run-in-sandbox.
If you have other ideas on how you would like to have this implemented, than go ahead and tell us please

**Reply by obuolinis** [reply]
If it's just a custom shared folder you need, I'd suggest to simply tweak RunInSandbox.ps1 in your installation, the part where wsb file is being generated.

**Reply by e-t-l** [reply]
I'm  trying to do this and I'm not seeing a clear way to make it work. I try adding another mapped folder in the config but it doesn't appear when I next launch a new Sandbox. Could someone provide an example of what specifically to add to the ps1 file in order to map a folder or install an app into the Sandbox?

**Reply by obuolinis** [reply]
To add an additional mapped folder you need to edit RunInSandbox.ps1 - inside it the function Generate_WSB is the one that does the job, you need to tweak it.

Start with reading Microsoft documentation on the Sandbox configuration file, specifically section about Mapped folders here:
https://learn.microsoft.com/en-us/windows/security/application-security/application-isolation/windows-sandbox/windows-sandbox-configure-using-wsb-file#mapped-folders

A corresponding section within Generate_WSB fucntion starts with line 116:
https://github.com/damienvanrobaeys/Run-in-Sandbox/blob/be67c90c43414b7facd5f0bdc4b7156d0e321d37/Sources/Run_in_Sandbox/RunInSandbox.ps1#L116
...and ends with line 150:
https://github.com/damienvanrobaeys/Run-in-Sandbox/blob/be67c90c43414b7facd5f0bdc4b7156d0e321d37/Sources/Run_in_Sandbox/RunInSandbox.ps1#L150

What you need to do is to add an additional "MappedFolder" block just before that line 150. It should look similar to this:
```powershell
Add-Content $Sandbox_File_Path "		<MappedFolder>"
Add-Content $Sandbox_File_Path "			<HostFolder>C:\Host\Folder</HostFolder>"
Add-Content $Sandbox_File_Path "			<SandboxFolder>C:\Sandbox\Folder</SandboxFolder>"
Add-Content $Sandbox_File_Path "			<ReadOnly>true</ReadOnly>"
Add-Content $Sandbox_File_Path "		</MappedFolder>"
```

**Comment by Joly0** [comment]
Hey guys, i am thinking about tackling this feature. If anyone is willing to create a feature request over at my fork https://github.com/Joly0/Run-in-Sandbox and add a compehensive explanation about what they want or wish this feature could provide, then i might be able to get it done.
